State.cs

```diff
@@ -289,19 +299,32 @@ public void Exit()
         /// current state is at the top of the stack, otherwise triggers it on 
         /// the next state down.
         /// </summary>
+        /// <param name="name">Name of the event to trigger</param>
         public void TriggerEvent(string name)
         {
+            TriggerEvent(name, null);
```

Comment:
Should pass in `EventArgs.Empty` here instead of `null`. Don't ever pass in a null unless there is no other sensible way to do it... use of null leads to `NullReferenceException`.